tests/test_apply.py

```diff
@@ -59,379 +59,255 @@
-    ("T", T),
-    ("Tdg", T.conj().T),
+    (qml.PauliX(wires=0), X),
+    (qml.PauliY(wires=0), Y),
+    (qml.PauliZ(wires=0), Z),
+    (qml.Hadamard(wires=0), H),
+    (qml.S(wires=0), S),
+    (qml.T(wires=0), T),
+    (qml.PauliX(wires=0).inv(), X.conj().T),
+    (qml.PauliY(wires=0).inv(), Y.conj().T),
+    (qml.PauliZ(wires=0).inv(), Z.conj().T),
+    (qml.Hadamard(wires=0).inv(), H.conj().T),
+    (qml.S(wires=0).inv(), S.conj().T),
+    (qml.T(wires=0).inv(), T.conj().T),
 ]
 
 # list of all parametrized single-qubit gates
-single_qubit_param = [("PhaseShift", phase_shift), ("RX", rx), ("RY", ry), ("RZ", rz)]
+single_qubit_param = [
+    (qml.RX(0, wires=0), rx),
+    (qml.RY(0, wires=0), ry),
+    (qml.RZ(0, wires=0), rz),
+    (qml.PhaseShift(0, wires=0), phase_shift),
+    (qml.RX(0, wires=0).inv(), lambda theta: rx(-theta)),
+    (qml.RY(0, wires=0).inv(), lambda theta: ry(-theta)),
+    (qml.RZ(0, wires=0).inv(), lambda theta: rz(-theta)),
+    (qml.PhaseShift(0, wires=0).inv(), lambda theta: phase_shift(-theta)),
+]
 # list of all non-parametrized two-qubit gates
-two_qubit = [("CNOT", CNOT), ("SWAP", SWAP), ("CZ", CZ)]
+two_qubit = [
+    (qml.CNOT(wires=[0, 1]), CNOT),
+    (qml.SWAP(wires=[0, 1]), SWAP),
+    (qml.CZ(wires=[0, 1]), CZ),
+    (qml.CNOT(wires=[0, 1]).inv(), CNOT.conj().T),
+    (qml.SWAP(wires=[0, 1]).inv(), SWAP.conj().T),
+    (qml.CZ(wires=[0, 1]).inv(), CZ.conj().T)
+]
 # list of all parametrized two-qubit gates
-two_qubit_param = [("CRZ", crz)]
+two_qubit_param = [
+    (qml.CRZ(0, wires=[0, 1]), crz),
+    (qml.CRZ(0, wires=[0, 1]).inv(), lambda theta: crz(-theta)),
+]
 # list of all three-qubit gates
-three_qubit = [("Toffoli", toffoli), ("CSWAP", CSWAP)]
-
-
-def reverse_state(state):
-    N = int(np.log2(len(state)))
-    return state.reshape([2] * N).T.flatten()
+three_qubit = [
+    (qml.Toffoli(wires=[0, 1, 2]), toffoli),
+    (qml.CSWAP(wires=[0, 1, 2]), CSWAP),
+    (qml.Toffoli(wires=[0, 1, 2]).inv(), toffoli.conj().T),
+    (qml.CSWAP(wires=[0, 1, 2]).inv(), CSWAP.conj().T),
+]
 
 
-@pytest.mark.parametrize("shots", [0])
 class TestStateApply:
-    """Test application of PennyLane operations to state simulators."""
-
-    def test_basis_state(self, device, tol):
+    """Test the device's state after application of gates."""
+
+    @pytest.mark.parametrize("state", [np.array([0, 0, 1, 0]),
+                                       np.array([0, 0, 1, 0]),
+                                       np.array([1, 0, 1, 0]),
+                                       np.array([1, 1, 1, 1])]
+                             )
+    def test_basis_state(self, state, tol):
         """Test basis state initialization"""
-        dev = device(4)
-        state = np.array([0, 0, 1, 0])
+        dev = QulacsDevice(4)
 
-        dev.apply("BasisState", [0, 1, 2, 3], [state])
+        op = qml.BasisState(state, wires=[0, 1, 2, 3])
+        dev.apply([op])
         dev._obs_queue = []
         dev.pre_measure()
```

Comment:
`QubitDevice` doesn't overwrite `pre_measure`, so here `Device.pre_measure` is being called which is an empty method. It is worth removing these so that once the `Device` class gets refactored (which we are considering), these tests will not break (a couple of further occurrences in this file).
```suggestion
```

Reply:
Done! Thanks @antalszava. Not sure why this is really here, though. What would be the reason for having `pre_measure()` called in these cases?

Reply:
Sure! It's due to a historical reason: the `QulacsDevice.pre_measure` method is being removed just now as part of this PR as it no longer has any function with `QulacsDevice` being subclassed from `QubitDevice`